Libraries/LibWeb/MimeSniff/Resource.cpp

```diff
@@ -183,6 +183,137 @@ bool matches_mp4_signature(ReadonlyBytes sequence)
     return false;
 }
 
+struct VintResult {
+    u64 value;
+    size_t length;
+};
+
+// https://mimesniff.spec.whatwg.org/#parse-a-vint
+VintResult parse_vint(ReadonlyBytes sequence, size_t iter)
+{
+    // To parse a vint on a byte sequence sequence of size length, starting at index iter use the following steps:
+    sequence = sequence.slice(iter);
+    auto length = sequence.size();
+
+    // 1. let mask be 128.
+    u8 mask = 128;
+
+    // 2. Let max vint length be 8.
+    size_t const max_vint_length = 8;
+
+    // 3. Let number size be 1.
+    size_t number_size = 1;
+
+    // 4. While number size is less than max vint length, and less than length, continuously loop through these steps:
+    while (number_size < max_vint_length && number_size < length) {
+        // 1. If the sequence[index] & mask is not zero, abort these steps.
+        if (sequence[0] & mask)
+            break;
+
+        // 2. Let mask be the value of mask >> 1.
+        mask >>= 1;
+
+        // 3. Increment number size by one.
+        number_size++;
+    }
+
+    // 5. Let index be 0.
+    size_t index = 0;
```

Comment:
The order of the steps here seems backwards, might be worth to create a spec issue for it at https://github.com/whatwg/mimesniff